From 0f7a4907117d55c953d0b0a6531088f7330e4b05 Mon Sep 17 00:00:00 2001
From: Hiep Pham <hiep.pham.zy@renesas.com>
Date: Thu, 22 Apr 2021 11:03:02 +0700
Subject: [PATCH 161/180] can: rcar-canfd: Add register Receive FIFO interrupt.

In RZ/G2L platform, we need register Receive FIFO interrupt
and Global Error interrupt as 2 individual interrupts sources.

Signed-off-by: Hiep Pham <hiep.pham.zy@renesas.com>
---
 drivers/net/can/rcar/rcar_canfd.c | 22 ++++++++++++++++++++--
 1 file changed, 20 insertions(+), 2 deletions(-)

diff --git a/drivers/net/can/rcar/rcar_canfd.c b/drivers/net/can/rcar/rcar_canfd.c
index 4d9efb7..aaefb4f 100644
--- a/drivers/net/can/rcar/rcar_canfd.c
+++ b/drivers/net/can/rcar/rcar_canfd.c
@@ -1701,7 +1701,7 @@ static int rcar_canfd_probe(struct platform_device *pdev)
 	struct rcar_canfd_global *gpriv;
 	struct device_node *of_child;
 	unsigned long channels_mask = 0;
-	int err, ch_irq, g_irq;
+	int err, ch_irq, g_irq, rx_irq;
 	bool fdmode = true;			/* CAN FD only mode - default */
 	void *data;
 	int i = 0;
@@ -1758,10 +1758,18 @@ static int rcar_canfd_probe(struct platform_device *pdev)
 	} else {
 		g_irq = platform_get_irq(pdev, 0);
 		if (g_irq < 0) {
-			dev_err(&pdev->dev, "no Global IRQ resource\n");
+			dev_err(&pdev->dev, "no Global error IRQ resource\n");
 			err = g_irq;
 			goto fail_dev;
 		}
+
+		rx_irq = platform_get_irq(pdev, 7);
+
+		if (rx_irq < 0) {
+			dev_err(&pdev->dev, "no Receive FIFO IRQ resource\n");
+			err = rx_irq;
+			goto fail_dev;
+		}
 	}
 
 	/* Peripheral clock */
@@ -1825,6 +1833,16 @@ static int rcar_canfd_probe(struct platform_device *pdev)
 		goto fail_dev;
 	}
 
+	err = devm_request_irq(&pdev->dev, rx_irq,
+			rcar_canfd_global_interrupt, 0,
+			"canfd.gbl", gpriv);
+
+	if (err) {
+		dev_err(&pdev->dev, "devm_request_irq(%d) failed, error %d\n",
+			rx_irq, err);
+		goto fail_dev;
+	}
+
 	/* Enable peripheral clock for register access */
 	err = clk_prepare_enable(gpriv->clkp);
 	if (err) {
-- 
2.7.4

